ui-testing-core/src/main/java/com/avito/android/test/espresso/action/recycler/RecyclerViewScroll.kt

```diff
@@ -0,0 +1,196 @@
+            position
+        )
+    }
+
+    override fun getConstraints(): Matcher<View> {
+        return Matchers.allOf<View>(
+            ViewMatchers.isAssignableFrom(RecyclerView::class.java),
+            ViewMatchers.isDisplayed()
+        )
+    }
+
+    override fun getDescription(): String {
+        return if (atPosition == RecyclerView.NO_POSITION) {
+            "scroll RecyclerView to: $viewHolderMatcher"
+        } else {
+            String.format(
+                "scroll RecyclerView to the: %dth matching %s.", atPosition,
+                viewHolderMatcher
+            )
+        }
+    }
+
+    override fun perform(uiController: UiController, view: View) {
+        val recyclerView = view as RecyclerView
+        try {
+            val maxMatches = if (atPosition == RecyclerView.NO_POSITION) 2 else atPosition + 1
+            val selectIndex = if (atPosition == RecyclerView.NO_POSITION) 0 else atPosition
+            val matchedItems = itemsMatching(
+                recyclerView,
+                viewHolderMatcher,
+                maxMatches
+            )
+
+            if (selectIndex >= matchedItems.size) {
+                throw RuntimeException(
+                    String.format(
+                        "Found %d items matching %s, but position %d was requested.",
+                        matchedItems.size,
+                        viewHolderMatcher.toString(),
+                        atPosition
+                    )
+                )
+            }
+            if (atPosition == RecyclerView.NO_POSITION && matchedItems.size == 2) {
+                val ambiguousViewError = StringBuilder()
+                ambiguousViewError.append(
+                    String.format("Found more than one sub-view matching %s", viewHolderMatcher)
+                )
+                for (item in matchedItems) {
+                    ambiguousViewError.append(item.toString() + "\n")
+                }
+                throw RuntimeException(ambiguousViewError.toString())
+            }
+            ScrollToPositionViewAction(matchedItems[selectIndex].position).perform(
+                uiController,
+                recyclerView
+            )
+            uiController.loopMainThreadUntilIdle()
+        } catch (e: RuntimeException) {
+            throw PerformException.Builder().withActionDescription(this.description)
+                .withViewDescription(HumanReadables.describe(view)).withCause(e).build()
+        }
+    }
+}
+
+fun <VH : RecyclerView.ViewHolder> scrollToHolder(
+    viewHolderMatcher: Matcher<VH>
+): RecyclerViewActions.PositionableRecyclerViewAction =
+    ScrollToViewAction(viewHolderMatcher)
+
+fun <VH : RecyclerView.ViewHolder> scrollTo(
+    itemViewMatcher: Matcher<View>
+): RecyclerViewActions.PositionableRecyclerViewAction {
+    val viewHolderMatcher = viewHolderMatcher<VH>(itemViewMatcher)
+    return ScrollToViewAction(viewHolderMatcher)
+}
+
+fun RecyclerView.scrollToPositionToCenter(
```

Comment:
All scroll actions for RecyclerView use this methods

Reply:
this and couple other functions and classes should be `internal`, right?